units/XNA0001/XNA0001_script.lua

```diff
@@ -1,36 +1,41 @@
 local NomadsEffectTemplate = import('/lua/nomadseffecttemplate.lua')
 local Unit = import('/lua/sim/Unit.lua').Unit
 
+---@class XNA0001 : Unit
 XNA0001 = Class(Unit) {
 
     CoverLaunchFx = NomadsEffectTemplate.ACUMeteorCoverLaunch,
     CoverOpenFx = NomadsEffectTemplate.ACUMeteorCoverOpen,
 
+    ---@param self XNA0001
     OnCreate = function(self)
+        local bp = self.Blueprint
+
         Unit.OnCreate(self)
 
-        self.EventCallbacks = table.merged(self.EventCallbacks, {
-            openup = {},
-        })
+        self.EventCallbacks = table.merged(self.EventCallbacks, {openup = {},})
 
-        if self:GetBlueprint().Display.AnimationOpen then
+        if bp.Display.AnimationOpen then
             self.OpenAnimManip = CreateAnimator(self)
             self.Trash:Add(self.OpenAnimManip)
-            self.OpenAnimManip:PlayAnim(self:GetBlueprint().Display.AnimationOpen, false):SetRate(0)
+            self.OpenAnimManip:PlayAnim(bp.Display.AnimationOpen, false):SetRate(0)
             self:OpenUp()
         else
             self:DelayedDestroy()
         end
     end,
 
+    ---@param self XNA0001
+    ---@param fn any
     AddOpenUpCallback = function(self, fn)
         -- fn -> function( <xna0001 instance>, <Opening anim state: opening|opened|expired>)
```

Comment:
[Annotations wiki docs for the `function` type](https://luals.github.io/wiki/annotations/#:~:text=VALUE_TYPE%2C%20key2%3A%20VALUE_TYPE%20%7D-,Function,fun(PARAM%3A%20TYPE)%3A%20RETURN_TYPE,-Unions%20may%20need)
```suggestion
    ---@param fn fun(instance: XNA0001, animationState: "opening" | "opened" | "expired")
    AddOpenUpCallback = function(self, fn)
```